Clear interceptors on configure #2

peteruithoven opened the pull request:
Currently when configure is called multiple times there will be multiple request interceptors, which usually means the checkRefresh is called multiple times uncecesarily. This also solves: #17

micky2be commented on src/index.js:
```diff
@@ -28,7 +28,6 @@ class Superlogin extends EventEmitter {
 		this._config = {};
 		this._refreshCB = this.refresh;
 		this._refreshInProgress = false;
-		this._http = axios.create();
```
Could you set `this._http` to `null` in the constructor please.

peteruithoven replied:
Sure, done.

peteruithoven commented:
@micky2be what do you think of it like this? Is this okay?

micky2be commented:
The fact that we have `getHttp` bother me, I don't wanna introduce a different bug.
I still believe that removing the event and putting them back would be better.

If you are not up for the task, give me some time so I can do it myself.

peteruithoven commented:
Which event are you talking about?

micky2be commented:
The interceptors.
They need to be cleaned with the API and re-attached during the setup.
This will guaranty everything to work as intended.

peteruithoven commented:
I'm not following at all...
Cleaned with what API? 
During which setup? Do you mean the configure() method of superlogin-client? 
Why clean something and then re-attach something? Are you talking about interseptors the user might ave added after using the getHttp method? 

Maybe we can, instead of creating a new axios instance, remove only the interseptors we've added?

micky2be commented:
Instead of re-instanciate Axios we could remove the interceptors and reattached it.
It will avoid people calling getHttp to use an old version.

peteruithoven commented:
True, like I was doing with the following pull request?
https://github.com/micky2be/superlogin-client/pull/18/commits/0c3cc0cfcdba0733589b5fcbed9426d9e0c05a7e

micky2be commented:
Yes

peteruithoven commented:
I'll revert back to that solution then.

micky2be commented:
I gave you tbe api to use last time. Why don't you use it?

’interceptors.request.eject‘

peteruithoven commented:
I explained why here: https://github.com/micky2be/superlogin-client/pull/18/commits/0c3cc0cfcdba0733589b5fcbed9426d9e0c05a7e#r76944725

peteruithoven commented:
I'm now storing the specific id's so we can eject them. This is probably what you meant? 
This has as added benefit that the user's added interceptors remain untouched.

micky2be commented:
yes, this looks better.
Thank you
